### ISSUE TITLE
File watching and `--open` broken on Windows

### ISSUE THREAD

**Issue by tomekwi** [tomekwi/elm-live#11]
As mentioned by @bdukes in https://github.com/tomekwi/elm-live/pull/10#issue-143414822 and https://github.com/tomekwi/elm-live/pull/10#issuecomment-201342084, we seem to have problems watching stuff and opening the browser.

**Comment by bdukes** [comment]
I've been doing some testing on this, and it appears to be related to _gaze_ taking forever to start watching all the files.  If I adjust the glob that's passed to avoid including `elm-stuff/`, then it takes about a minute to get started, otherwise I've left it running for tens of minutes without the server starting.  One small help would be to start gaze asynchronously, so it doesn't block the server starting when it's trying to find everything.  I'm not sure if there's a safer default glob to pass, maybe if it was possible to configure the glob (so you could point to a particular directory)?

**Comment by bdukes** [comment]
Looks like I exaggerated to myself a bit.  I added some timing, and it takes 7-8 minutes to start when `elm-stuff` is included.

**Comment by bdukes** [comment]
Are you attached to using _gaze_?  Since _budo_  has watching built in, I looked at what it's using, _[chokidar](https://github.com/paulmillr/chokidar)_, and it appears to work much better (probably mostly because it's async, not blocking).  I can submit a PR with that change, so long as there's not a specific reason you were using _gaze_.

**Comment by tomekwi** [comment]
Hi @bdukes, and thanks for all your investigation. I’m absolutely not attached to using *gaze* and I’ve seen other fine libs and programs use *chokidar*. I’ll have a look at how it works right away.

However, I’m a bit surprized the *gaze* call blocks. Have a look at [source/elm-live.js:111](https://github.com/tomekwi/elm-live/blob/6b02990f64352556692eb48d953a6ab37f315a9d/source/elm-live.js#L111) – the server should start right away, even before *gaze* starts looking for files – and *gaze* is an enchancement which should come later. Is the initial build successful? Perhaps your issue points at a deeper problem?

By the way, disabling `elm-stuff/` by default and letting people configure it may be the way forward. I didn’t want to exclude it without making it configurable, because I sometimes develop more than one project in parallel. And I didn’t want to make it configurable as long as it *just works*. But it seems like it doesn’t *just work* anymore. :wink:

**Comment by bdukes** [comment]
Based on what I've seen, the server doesn't start connecting until later (i.e. it's using asynchronous APIs).  Since the `gaze` call blocks, control isn't returned to the event loop to allow the server a chance to do the rest of its work.  The timing of it was that the first build _does_ succeed, and I could log the `server` instance created from calling `budo`, but the server wouldn't start up (nothing logged, no browser opened, etc.) until after the `gaze` call, which would take minutes to complete.

It does probably make sense for the watch glob to be configurable, but using chokidar, it does "just work" for me, so that's now a lower priority, from my perspective.
